contracts/staking/contracts/src/StakingProxy.sol

```diff
@@ -0,0 +1,109 @@
+/*
+
+  Copyright 2018 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.5;
+
+import "./immutable/MixinStorage.sol";
+import "./interfaces/IStakingProxy.sol";
+import "./sys/MixinOwnable.sol";
+
+
+contract StakingProxy is
+    IStakingProxy,
+    MixinDeploymentConstants,
+    MixinConstants,
+    MixinStorage,
+    MixinOwnable
+{
+
+    /// @dev Constructor.
+    /// @param _stakingContract Staking contract to delegate calls to.
+    constructor(address _stakingContract)
+        public
+    {
+        owner = msg.sender;
```

Comment:
Mm yeah this actually raises an interesting design decision that I'm curious to get your thoughts on: none of the mixins have constructors because they all share the proxy's state, which is not available during constructing of the staking contract. Their state is instead placed in `immutable/MixinStorage.sol` and inherited by both the proxy and the staking contract.